cli/cmd/utils/utils.go

```diff
@@ -160,6 +163,11 @@ func GetCLIContext(command *cobra.Command) (SoftwareFactoryConfigContext, error)
 	// Override with defaults
 	// We don't set a default namespace here so as not to interfere with rootcommand.
 	ns, _ := command.Flags().GetString("namespace")
+	currentContext := GetKubeConfigCurrentContext()
+	if ns == "" && currentContext != nil {
+		ns = currentContext.Namespace
+	}
+
```

Comment:
I think we need to grab the namespace name based on the kube-context provided by the user, see L174 below. So this block should be moved below, and if the cliContext.kubeContext is set, then use that L536. Otherwise we may use the wrong namespace if the kubeconfig has multiple namespaces.

Reply:
done, but this ways the user always has to set or namespace or kube-context, i thought it was to behave like the kubectl cli

Reply:
It could work like kubectl: if no context is provided, then use the default one, if no namespace is provided, then use the context default one, and if no namespace is found, raise an error to request a namespace param.